CMakeLists.txt

```diff
@@ -32,6 +32,11 @@ endif()
 
 option(ASAN "Enable AddressSanitizer (ASan)" OFF)
 option(TSAN "Enable ThreadSanitizer (TSan)" OFF)
+option(USE_OSX_ASYNC_LIBINFO "Enable async getnameinfo on OSX" ON)
+if(USE_OSX_ASYNC_LIBINFO AND APPLE)
```

Comment:
missing check for libinfo itself.

Reply:
@bnoordhuis AFAIU linkers do lazy-load by default these days, I did check and https://linux.die.net/man/1/ld says `-z lazy` is the default, it's only needed if you `-z now` to force it to resolve symbols immediately.

Going with `dlopen` is not that hard, but adds a bulk of code that could be avoided by using the correct flags.

Reply:
The reason we do that is to make life easier to use for projects linking statically against libuv. No need for the downstream project to link against libinfo, or CoreFoundation, or IOKit, or...

Reply:
but I think detecting the library with https://cmake.org/cmake/help/latest/command/find_library.html is the way to go to avoid misusage

Reply:
To clarify: using dlopen() is a requirement.

Reply:
sure, with misusage I meant trying to build this on non-mac systems... but on macOS would be nice to have that enabled by default (thus checking helps there)